src/librbd/io/CopyupRequest.cc

```diff
@@ -206,10 +208,31 @@ bool CopyupRequest<I>::is_copyup_required() {
   return false;
 }
 
+template <typename I>
+bool CopyupRequest<I>::is_deep_copy() const {
+  return m_ictx->parent_md.spec.migration_source &&
+    !m_ictx->parent_md.spec.snap_seqs.empty();
+}
+
 template <typename I>
 void CopyupRequest<I>::send()
 {
   m_state = STATE_READ_FROM_PARENT;
+
+  if (is_deep_copy()) {
+    deep_copy::util::compute_snap_map(0, CEPH_NOSNAP,
```

Comment:
Nit: this can probably just be pre-computed within the ```RefreshRequest``` state machine instead of storing the ```snap_seqs``` in memory